pkg/odo/cli/storage/create.go

```diff
@@ -24,49 +25,98 @@ var (
 	`)
 )
 
+const defaultStorageSize = "1Gi"
+
 type StorageCreateOptions struct {
 	storageName      string
 	storageSize      string
 	storagePath      string
 	componentContext string
+
+	devfilePath   string
+	isDevfile     bool
+	componentName string
 	*genericclioptions.Context
 }
 
 // NewStorageCreateOptions creates a new StorageCreateOptions instance
 func NewStorageCreateOptions() *StorageCreateOptions {
-	return &StorageCreateOptions{}
+	return &StorageCreateOptions{devfilePath: "./devfile.yaml"}
 }
 
 // Complete completes StorageCreateOptions after they've been created
 func (o *StorageCreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	o.Context = genericclioptions.NewContext(cmd)
+	if o.isDevfile {
+		o.Context = genericclioptions.NewDevfileContext(cmd)
+
+		o.componentName = o.EnvSpecificInfo.GetName()
+		if o.storageSize == "" {
+			o.storageSize = defaultStorageSize
+		}
+	} else {
+		o.Context = genericclioptions.NewContext(cmd)
+		o.componentName = o.LocalConfigInfo.GetName()
+	}
+
 	if len(args) != 0 {
 		o.storageName = args[0]
 	} else {
-		o.storageName = o.Component() + "-" + util.GenerateRandomString(4)
+		o.storageName = o.componentName + "-" + util.GenerateRandomString(4)
```

Comment:
may be better to do `fmt.Springf(%s-%d), o.componentName, util.GenerateRandomString(4)` similar to my other comment.